homeassistant/const.py

```diff
@@ -487,8 +487,9 @@ class Platform(StrEnum):
 POWER_VOLT_AMPERE_REACTIVE: Final = "var"
 
 # Energy units
-ENERGY_WATT_HOUR: Final = "Wh"
+ENERGY_GIGA_JOULE: Final = "GJ"
 ENERGY_KILO_WATT_HOUR: Final = "kWh"
+ENERGY_WATT_HOUR: Final = "Wh"
 ENERGY_MEGA_WATT_HOUR: Final = "MWh"
```

Comment:
I think the PR as a whole looks good. Just two minor comments:

Were you trying to have them in alphabetical order?
```suggestion
ENERGY_MEGA_WATT_HOUR: Final = "MWh"
ENERGY_WATT_HOUR: Final = "Wh"
```

Also, would it make sense to introduce a new `StrEnum` for the energy units, instead of adding `GJ` as new standalone constant?

```python
class UnitOfEnergy(StrEnum):
    """Energy units."""

    GIGA_JOULE = "GJ"

    KILO_WATT_HOUR = "kWh"

    MEGA_WATT_HOUR = "MWh"

    WATT_HOUR = "Wh"
```

Reply:
The enum can be considered in a separate PR

Reply:
I opened https://github.com/home-assistant/core/pull/80998, dropping the new constant and replacing it with an enum.